tests/bench_sizeof_coretypes/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2014 René Kijewski <rene.kijewski@fu-berlin.de>
+ *               2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Print size of core types
+ *
+ * @author      René Kijewski <rene.kijewski@fu-berlin.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "cib.h"
+#include "clist.h"
+#include "panic.h"
+#include "kernel_types.h"
+#include "list.h"
+#include "mbox.h"
+#include "msg.h"
+#include "mutex.h"
+#include "priority_queue.h"
+#include "ringbuffer.h"
+#include "rmutex.h"
+#ifdef MODULE_CORE_THREAD_FLAGS
+#include "thread_flags.h"
+#endif
+#include "thread.h"
+
+
+#define P(NAME) printf("    tcb->%-11s            %3u     %3u\n", #NAME, \
+                       (unsigned)sizeof(((thread_t *) 0)->NAME), \
+                       (unsigned)offsetof(thread_t, NAME));
+
+int main(void)
+{
+    puts("Sizeof RIOT core types\n");
+
+    puts("                                size");
+
+    printf("sizeof(cib_t):                  %3u\n", sizeof(cib_t));
+    printf("sizeof(clist_node_t):           %3u\n", sizeof(clist_node_t));
+    printf("sizeof(core_panic_t):           %3u\n", sizeof(core_panic_t));
+    printf("sizeof(kernel_pid_t):           %3u\n", sizeof(kernel_pid_t));
+    printf("sizeof(list_node_t):            %3u\n", sizeof(list_node_t));
+    printf("sizeof(mbox_t):                 %3u\n", sizeof(mbox_t));
+#ifdef MODULE_CORE_MSG
+    printf("sizeof(msg_t):                  %3u\n", sizeof(msg_t));
+#else
+    puts("sizeof(msg_t):                    0   (not enabled)");
+#endif
+    printf("sizeof(mutex_t):                %3u\n", sizeof(mutex_t));
+    printf("sizeof(priority_queue_node_t):  %3u\n", sizeof(priority_queue_node_t));
+    printf("sizeof(priority_queue_t):       %3u\n", sizeof(priority_queue_t));
+    printf("sizeof(ringbuffer_t):           %3u\n", sizeof(ringbuffer_t));
+    printf("sizeof(rmutex_t):               %3u\n", sizeof(rmutex_t));
+#ifdef MODULE_CORE_THREAD_FLAGS
+    printf("sizeof(thread_flags_t):         %3u\n", sizeof(thread_flags_t));
+#else
+    puts("sizeof(thread_flags_t):           0   (not enabled)");
+#endif
+    printf("\nTCB (thread_t) details:         size  offset\n");
+    printf("sizeof(thread_t):               %3u       -\n", sizeof(thread_t));
```

Comment:
Could all of these go in a macro? Could have two macros, `P_TCB(NAME)` (a rename of P(NAME) above) and `P(NAME)` (to handle non-TCB sizeof() printouts)? Also, could these two macros go in benchmark.h to allow re-use?

Reply:
could, but does it really make things better readable? The current `P(NAME)` macro is quite ugly and only there because I copy-pasted it from the old application. IMHO having the printfs makes thinks quite nice to read and is pretty much straight forward... But this is subjective and the kind of thing I don't really see its worth spending time on...